sqlsynthgen/utils.py

```diff
@@ -51,13 +52,24 @@ def download_table(table: Any, engine: Any, yaml_file_name: str) -> None:
         yamlfile.write(yaml.dump(result))
 
 
-def create_engine_with_search_path(postgres_dsn: PostgresDsn, schema_name: str) -> Any:
-    """Create a SQLAlchemy Engine with an explicitly set schema."""
-    engine = create_engine(postgres_dsn)
-
-    @event.listens_for(engine, "connect", insert=True)
-    def connect(dbapi_connection: Any, _: Any) -> None:
-        set_search_path(dbapi_connection, schema_name)
+def create_db_engine(
+    postgres_dsn: PostgresDsn,
+    schema_name: Optional[str] = None,
+    use_asyncio: bool = False,
+    **kwargs: dict,
+) -> Any:
+    """Create a SQLAlchemy Engine."""
+    if use_asyncio:
+        async_dsn = postgres_dsn.replace("postgresql://", "postgresql+asyncpg://")
+        engine = create_async_engine(async_dsn, **kwargs)
+    else:
+        engine = create_engine(postgres_dsn, **kwargs)
+
+    if schema_name is not None:
+
+        @event.listens_for(engine, "connect", insert=True)
+        def connect(dbapi_connection: Any, _: Any) -> None:
+            set_search_path(dbapi_connection, schema_name)  # type: ignore
```

Comment:
The schema event breaks if you try to attach it to an async engine.

```suggestion
        engine = create_async_engine(async_dsn, **kwargs)
        event_engine = engine.sync_engine
    else:
        engine = create_engine(postgres_dsn, **kwargs)
        event_engine = engine
        
    if schema_name is not None:

        @event.listens_for(event_engine, "connect", insert=True)
        def connect(dbapi_connection: Any, _: Any) -> None:
            set_search_path(dbapi_connection, schema_name)  # type: ignore
```

Reply:
See #101